lib/dotcom_web/components/trip_planner/transit_leg.ex

```diff
@@ -172,4 +182,31 @@ defmodule DotcomWeb.Components.TripPlanner.TransitLeg do
     </ul>
     """
   end
+
+  defp group_alerts(alerts, leg) when is_list(alerts) do
+    grouped_alerts =
+      alerts
+      |> Enum.group_by(fn alert ->
+        alert.informed_entity.entities
+        |> Enum.any?(fn
+          %{stop: nil} -> false
+          _ -> true
+        end)
+      end)
+
+    route_alerts = grouped_alerts[false] || []
+    stop_alerts = grouped_alerts[true] || []
+
+    entities_from = Alerts.leg_entities_from(leg)
+    alerts_for_from = Match.match(stop_alerts, entities_from)
+
+    entities_to = Alerts.leg_entities_to(leg)
+    alerts_for_to = Match.match(stop_alerts, entities_to)
+
+    %{
+      alerts_for_mode: route_alerts,
+      alerts_for_from: alerts_for_from,
+      alerts_for_to: alerts_for_to
+    }
```

Comment:
There _has_ to be a better way to do this.

I'm pretty 👎 on both the actual logic here and the location - I only did it like this to start in order to prove out the overall concept. I think longer-term (maybe even before merging this PR), alerts should be passed into this component grouped correctly.

I'm going to hold off on putting this into the actually-correct location until https://github.com/mbta/dotcom/pull/2266 is merged, because that PR might inform how alerts state should be handled.

I also couldn't see an easy way (aside from just checking that an alert's "informed entities" didn't have any stops) to tell whether an alert was for a route or for a stop.

Reply:
At minimum I'd suggest moving this function to a public function in `Dotcom.TripPlan.Alerts`, and IMO the keys don't need the `alerts_for_` prefix 🤔

Reply:
[Moved](https://github.com/mbta/dotcom/pull/2269/commits/4e327b25650d5635b2bc115fdca721e0d874d29c) and [renamed](https://github.com/mbta/dotcom/pull/2269/commits/8225f4e0b34d40f780b9a4da42141f7b57b05f8b)!